packages/protocol/contracts/L1/libs/LibProving.sol

```diff
@@ -148,70 +82,71 @@ library LibProving {
 
             if (fc.prover != address(0)) revert L1_ALREADY_PROVEN();
 
-            fc.prover = prover;
+            fc.prover = evidence.prover;
             fc.provenAt = uint64(block.timestamp);
         }
 
         if (!oracleProving && !config.skipZKPVerification) {
             // Do not revert when circuitId is invalid.
             address verifier = resolver.resolve(
                 string.concat(
-                    snippet.blockHash == LibUtils.BLOCK_DEADEND_HASH
-                        ? "vib" // verifier for invalid blocks
-                        : "vb", // verifier for valid blocks
-                    Strings.toString(zkproof.circuitId)
+                    "plonk_verifier",
+                    Strings.toString(evidence.zkproof.circuitId)
                 ),
                 true
             );
             if (verifier == address(0)) revert L1_INVALID_PROOF();
 
-            (bool verified, ) = verifier.staticcall(
+            bytes32 instance;
+            if (evidence.blockHash == LibUtils.BLOCK_DEADEND_HASH) {
+                if (evidence.signalRoot != 0) revert L1_NONZERO_SIGNAL_ROOT();
+                instance = evidence.meta.txListHash;
+            } else {
+                address l1SignalService = resolver.resolve(
+                    "signal_service",
+                    false
+                );
+                address l2SignalService = resolver.resolve(
+                    config.chainId,
+                    "signal_service",
+                    false
+                );
+
+                instance = keccak256(
```

Comment:
I think data like `timestamp` and `mixhash` are missing here so the circuit can verify the blockhash is calculated using these expected values. Or could also be I'm overlooking where these are inserted in the circuit.

Reply:
Brecht, since we no longer pass L2 block header to the `proveBlock` function, we don't have access to the `timestamp` and `mixhash` values on L1.